### PR TITLE
[LPTOCPCI-548] Allow to specify priority and assignee

### PR DESCRIPTION
**Pull request by calebevans**
**Modifications**
- **[70081a9](https://github.com/CSPI-QE/firewatch/pull/23/commits/70081a9c967bf6fdfc0324ec21c5775f866da18a)**: Slightly modified the way the firewatch configuration is handled. I noticed that there wasn't any error handling or validation around obtaining the default project or the firewatch config.

**New Features**
- **[fc28391](https://github.com/CSPI-QE/firewatch/pull/23/commits/fc28391df98feee7eef38873f3d6119f839cadf6)**: I have added the new optional field `jira_assignee`. This field allows users to specify which Jira user bugs should be filed under. The value must be the email address of the user they would like to assign bugs to.
- **[b9ffee5](https://github.com/CSPI-QE/firewatch/pull/23/commits/b9ffee5513c7a5acbb57b79c7155776a3291239e)**: I have added the new optional field `jira_priority` This field allows users to specify the priority that bugs should be set to when filed for each rule. This value must be one of the following:
  - `Blocker`
  - `Critical`
  - `Major`
  - `Normal`
  - `Minor`

### CODE REVIEW

File: cli/objects/rule.py [hunk 1]
```diff
@@ -274,6 +276,68 @@ def _get_jira_additional_labels(
...
+        :return: A string of the Jira assignee to use in a firewatch rule. If one is not defined, return None
+        """
+        if "jira_assignee" in rule_dict.keys():
+            try:
+                jira_assignee = rule_dict["jira_assignee"]
+            except Exception as ex:
+                self.logger.error(ex)
+                exit(1)
+            if isinstance(jira_assignee, str):
+                if "@" in jira_assignee:
```
Comment (Simranpal) [inline cli/objects/rule.py:293]: You can check for `"@redhat.com"` here

**Review by Simranpal — COMMENTED** [review]
Looks good, just minor suggestion on email validation

File: cli/objects/rule.py (hunk 1 above)
Comment (calebevans) [reply cli/objects/rule.py:293]: @Simranpal I considered this, but wasn't sure if we have users in Jira that have different domains in their email (IBM maybe?)

Comment (calebevans) [reply cli/objects/rule.py]: @Simranpal how is this: [c56a85f](https://github.com/CSPI-QE/firewatch/pull/23/commits/c56a85fed24aa9c19101f22d6ecc5f19695ec4d4)

Comment (Simranpal) [reply cli/objects/rule.py]: yes looks good @calebevans

File: cli/objects/rule.py [hunk 2]
```diff
@@ -274,6 +276,68 @@ def _get_jira_additional_labels(
...
+    def _get_jira_assignee(self, rule_dict: dict[Any, Any]) -> Optional[str]:
+        """
+        Determines if a Jira Assignee is defined in a rule. If it is, validate it and return the string.
+
+        :param rule_dict: A dictionary object representing a user-defined firewatch rule.
+        :return: A string of the Jira assignee to use in a firewatch rule. If one is not defined, return None
+        """
+        if "jira_assignee" in rule_dict.keys():
+            try:
+                jira_assignee = rule_dict["jira_assignee"]
```
Comment (btjd) [inline cli/objects/rule.py:289]: Nitpick, as far as I know, I think an exception would only get thrown here is the key didn't exist in the dict, and that's handled by line 286, so the use of try catch here might be unnecessary unless there is a corner case I am not thinking of :)

**Review by Simranpal — APPROVED** [review]

File: cli/objects/rule.py [hunk 3]
```diff
@@ -274,6 +276,68 @@ def _get_jira_additional_labels(
...
+            except Exception as ex:
+                self.logger.error(ex)
+                exit(1)
+            if isinstance(jira_assignee, str):
```
Comment (btjd) [inline cli/objects/rule.py:293]: Would it be better to use a regex here?

File: cli/objects/configuration.py [hunk 4]
```diff
@@ -88,3 +76,42 @@ def _get_rules(self, rules_json: list[dict[Any, Any]]) -> Optional[list[Rule]]:
...
+        if default_project:
+            return default_project
+        else:
+            self.logger.error(
+                "Environment variable $FIREWATCH_DEFAULT_JIRA_PROJECT is not set, please set the variable and try again.",
+            )
+            exit(1)
+
+    def _get_config_data(self, config_file_path: Optional[str]) -> str:
+        """"""
```
Comment (ryankwilliams) [inline cli/objects/configuration.py:97]: Just an FYI, missing docstring here.

File: cli/objects/rule.py [hunk 5]
```diff
@@ -274,6 +277,69 @@ def _get_jira_additional_labels(
...
+        """
+        Determines if a Jira priority is defined in a rule. If it is, validate it and return the string.
+
+        :param rule_dict: A dictionary object representing a user-defined firewatch rule.
+        :return: A string of the Jira priority to use in a firewatch rule. If one is not defined, return None
+        """
+        valid_priority_values = ["Blocker", "Critical", "Major", "Normal", "Minor"]
+
+        if "jira_priority" in rule_dict.keys():
+            try:
```
Comment (ryankwilliams) [inline cli/objects/rule.py:320]: Similar to the comment here: https://github.com/CSPI-QE/firewatch/pull/23/files#r1275267525

File: cli/objects/rule.py (hunk 2 above)
Comment (calebevans) [reply cli/objects/rule.py:289]: You are correct, my bad! Will update in an upcoming commit.

File: cli/objects/rule.py (hunk 3 above)
Comment (calebevans) [reply cli/objects/rule.py:293]: Looks like we had the same thought :) https://github.com/CSPI-QE/firewatch/pull/23#discussion_r1275264696

File: cli/objects/rule.py (hunk 5 above)
Comment (calebevans) [reply cli/objects/rule.py:320]: Will fix in an upcoming commit!

File: cli/objects/configuration.py (hunk 4 above)
Comment (calebevans) [reply cli/objects/configuration.py]: Fixed in [dc163bd](https://github.com/CSPI-QE/firewatch/pull/23/commits/dc163bdcb6400f16ab207376c34d4f04f83193e4). Thanks!

File: cli/objects/rule.py (hunk 2 above)
Comment (calebevans) [reply cli/objects/rule.py]: Fixed in [b819732](https://github.com/CSPI-QE/firewatch/pull/23/commits/b819732d735899bd815b3292b116c2abf2d3aa67). Thanks!

File: cli/objects/rule.py (hunk 5 above)
Comment (calebevans) [reply cli/objects/rule.py]: Fixed in [b819732](https://github.com/CSPI-QE/firewatch/pull/23/commits/b819732d735899bd815b3292b116c2abf2d3aa67). Thanks!

**Pull request merged**